### PR TITLE
[Bug](sink) update error status on BufferControlBlock::close

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
update error status on BufferControlBlock::close

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/runtime/buffer_control_block.cpp [hunk 1]
```diff
@@ -248,6 +248,10 @@ Status BufferControlBlock::get_arrow_batch(std::shared_ptr<arrow::RecordBatch>*
 
 Status BufferControlBlock::close(const TUniqueId& id, Status exec_status) {
     std::unique_lock<std::mutex> l(_lock);
+    if (!exec_status.ok()) {
```
Comment (yiguolei) [inline be/src/runtime/buffer_control_block.cpp:251]: add comment here to explain the reason.

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by zhiqiang-hhhh — APPROVED** [review]

**Pull request merged**